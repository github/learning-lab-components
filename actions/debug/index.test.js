const debug = require('./')
const mockContext = require('../../tests/mockContext')

describe('debug', () => {
  let context

  beforeEach(() => {
    context = mockContext({}, {
      repos: {
        createHook: jest.fn()
      }
    })
  })

  it('sets the repo webhook for a draft course version', async () => {
    context.courseVersion.draft = true
    await debug(context, {})
    expect(context.github.repos.createHook).toHaveBeenCalled()
    expect(context.github.repos.createHook.mock.calls).toMatchSnapshot()
  })

  it('does not set the repo webhook for a non-draft course version', async () => {
    context.courseVersion.draft = false
    await debug(context, {})
    expect(context.github.repos.createHook).not.toHaveBeenCalled()
  })
})
