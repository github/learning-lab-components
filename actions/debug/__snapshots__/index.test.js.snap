// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`debug sets the repo webhook for a draft course version 1`] = `
Array [
  Array [
    Object {
      "config": Object {
        "content_type": "json",
        "events": Array [
          "push",
          "issues",
          "pull_request",
          "issue_comment",
        ],
        "url": "https://smee.io/undefined-JasonEtco",
      },
      "name": "web",
      "owner": "JasonEtco",
      "repo": "example",
    },
  ],
]
`;
